### PR TITLE
Homework

### PR DESCRIPTION
**Pull request by ShangTsing**

### CODE REVIEW

**Comment by iphysresearch** [comment]
Good Job!!!   @ShangTsing
---

**Comment by iphysresearch** [comment]
Good Job!!!   @ShangTsing
---

**Comment by iphysresearch** [comment]
> ## Good Job!!! @ShangTsing

@ShangTsing 
一般这种问题是由于没有 update branch 一下你的homework分支，导致有不止一个作业文件在modified状态。 https://github.com/iphysresearch/GWData-Bootcamp/pull/137#issuecomment-1845666444

**Comment by iphysresearch** [comment]
Good Job!!!   @ShangTsing
---
Python homework:
- Total questions: 108
- Correct answers: 105
- Score: 97.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---

**Comment by iphysresearch** [comment]
Good Job!!!   @ShangTsing
---
Python homework:
- Total questions: 108
- Correct answers: 105
- Score: 97.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---

**Comment by iphysresearch** [comment]
Good Job!!!   @ShangTsing
---
Python homework:
- Total questions: 108
- Correct answers: 107
- Score: 99.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---

**Comment by iphysresearch** [comment]
Good Job!!!   @ShangTsing
---
Python homework:
- Total questions: 108
- Correct answers: 106
- Score: 98.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---

**Comment by iphysresearch** [comment]
Good Job!!!   @ShangTsing
---
Python homework:
- Total questions: 108
- Correct answers: 106
- Score: 98.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---

**Comment by iphysresearch** [comment]
Good Job!!!   @ShangTsing
---
Python homework:
- Total questions: 108
- Correct answers: 106
- Score: 98.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---

**Comment by iphysresearch** [comment]
Good Job!!!   @ShangTsing
---
Python homework:
- Total questions: 108
- Correct answers: 105
- Score: 97.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---

**Comment by iphysresearch** [comment]
Good Job!!!   @ShangTsing
---
Python homework:
- Total questions: 108
- Correct answers: 106
- Score: 98.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---

**Comment by iphysresearch** [comment]
Good Job!!!   @ShangTsing
---
Python homework:
- Total questions: 108
- Correct answers: 107
- Score: 99.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---

**Comment by iphysresearch** [comment]
Good Job!!!   @ShangTsing
---
Python homework:
- Total questions: 108
- Correct answers: 107
- Score: 99.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---
Pandas homework:
- Total questions: 12
- Correct answers: 10
- Score: 83.00%
---

**Comment by iphysresearch** [comment]
Good Job!!!   @ShangTsing
---
Python homework:
- Total questions: 108
- Correct answers: 107
- Score: 99.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---
Pandas homework:
- Total questions: 12
- Correct answers: 11
- Score: 91.00%
---

**Comment by iphysresearch** [comment]
Good Job!!!   @ShangTsing
---
Python homework:
- Total questions: 108
- Correct answers: 107
- Score: 99.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---
Pandas homework:
- Total questions: 12
- Correct answers: 12
- Score: 100.00%
---

**Comment by iphysresearch** [comment]
Good Job!!!   @ShangTsing
---
Python homework:
- Total questions: 108
- Correct answers: 108
- Score: 100.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---
Pandas homework:
- Total questions: 12
- Correct answers: 12
- Score: 100.00%
---

**Comment by ShangTsing** [comment]
@iphysresearch 老师好，这是我的Python、Numpy 和 Pandas 的基础作业，麻烦记录一下成绩吧，其他作业我会尽量在截止前完成，谢谢老师！

**Review by iphysresearch — APPROVED** [review]
@ShangTsing 
很好哈，把Python编程的扩展作业也做一下吧哈？

**Comment by iphysresearch** [comment]
Good Job!!!   @ShangTsing
---
Python homework:
- Total questions: 108
- Correct answers: 108
- Score: 100.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---
Pandas homework:
- Total questions: 12
- Correct answers: 12
- Score: 100.00%
---

**Comment by iphysresearch** [comment]
Good Job!!!   @ShangTsing
---
Python homework:
- Total questions: 108
- Correct answers: 108
- Score: 100.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---
Pandas homework:
- Total questions: 12
- Correct answers: 12
- Score: 100.00%
---

File: 2023/homework/CaoXu/homework-matplotlib_seaborn.ipynb
Comment (iphysresearch) [inline 2023/homework/CaoXu/homework-matplotlib_seaborn.ipynb:1]: Matplotlib数据可视化作业！

**Review by iphysresearch — APPROVED** [review]
@ShangTsing 
完成Python基础作业+Matplotlib数据可视化作业！

**Comment by ShangTsing** [comment]
```
def checkSegments(s):
    longest_1s = 0
    longest_0s = 0
    current_1s = 0
    current_0s = 0

    for char in s:
        if char == '1':
            current_1s += 1
            current_0s = 0
            longest_1s = max(longest_1s, current_1s)
        else:
            current_0s += 1
            current_1s = 0
            longest_0s = max(longest_0s, current_0s)

    return longest_1s > longest_0s

# Example usage:
s = "1101"
result = checkSegments(s)
print(result)  # Output: True

s = "111000"
result = checkSegments(s)
print(result)  # Output: False

s = "110100010"
result = checkSegments(s)
print(result)  # Output: False
```

**Comment by ShangTsing** [comment]
`def checkSegment(s):
    count = 0

    for i in range(1, len(s)):
        if s[i] == '1':
            count += 1
            if count > 1:
                return False
        else:
            count = 0

    return True`

**Comment by ShangTsing** [comment]
```
def findPeakElement(arr):
    left, right = 0, len(arr) - 1

    while left < right:
        mid = left + (right - left) // 2

        if arr[mid] < arr[mid + 1]:
            left = mid + 1
        else:
            right = mid

    return left
```

**Comment by ShangTsing** [comment]
```
def findPeakElement(nums):

   #初始化两个数，为输入nums的第一个数与最后一个数的index
    left, right = 0, len(nums) - 1
  #循环，不断取中间值mid 
    while left < right:
        mid = left + (right - left) // 2
  #重新定义left和right知道找到最大值的指标
        if nums[mid] < nums[mid + 1]:
            left = mid + 1
        else:
            right = mid

    return left
```

**Comment by ShangTsing** [comment]
@iphysresearch 老师这是我的Python拓展作业，机器学习的作业刚刚在工位上还在跑最后一题，回到寝室发现远程软件忘开了😭，其他的作业也没时间做了，今天应该只能交这些作业了，对不起老师。

**Pull request merged**